libraries/core_libs/network/src/network.cpp

```diff
@@ -37,16 +40,18 @@ Network::Network(NetworkConfig const &config, std::filesystem::path const &netwo
   taraxa_net_conf.expected_parallelism = tp_.capacity();
 
   string net_version = "TaraxaNode";  // TODO maybe give a proper name?
-  auto construct_capabilities = [&, this](auto host) {
-    assert(!host.expired());
-
-    taraxa_capability_ = std::make_shared<network::tarcap::TaraxaCapability>(host, key, conf_, db, pbft_mgr, pbft_chain,
-                                                                             vote_mgr, next_votes_mgr, dag_mgr,
-                                                                             dag_blk_mgr, trx_mgr, key.address());
-    return dev::p2p::Host::CapabilityList{taraxa_capability_};
-  };
+  if (!construct_capabilities) {
+    construct_capabilities = [&](std::weak_ptr<dev::p2p::Host> host) {
+      assert(!host.expired());
+
+      auto taraxa_capability =
+          network::tarcap::TaraxaCapability::make(host, key, conf_, TARAXA_NET_VERSION, db, pbft_mgr, pbft_chain,
```

Comment:
So how does this static `make` method together with `init` method makes it any different than calling `registerPacketHandlers` & `initPeriodicEvents` directly in constructor ?

Reply:
ok, in this case first part of my comment still applies: 
So how does this static make method together with init method makes it any different than calling virtual registerPacketHandlers & initPeriodicEvents directly in constructor ?